File: netease/src/main/java/com/one/netease/MainActivity.java
package com.one.netease;

import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tv = findViewById(R.id.tv);

        Handler handler = new Handler(){
            @Override
            public void handleMessage(@NonNull Message msg) {
                super.handleMessage(msg);
            }
        };

        Handler handler1 = new Handler(new Handler.Callback() {
            @Override
            public boolean handleMessage(@NonNull Message msg) {
                switch (msg.what) {
                    case 1:

                        String str = (String) msg.obj;

                        tv.setText(str);
                        break;
                }
                return false;
            }
        });


        new Thread(new Runnable() {
            @Override
            public void run() {
                Message msg = new Message();
                msg.what=1;
                msg.obj = "hello onebit";

                try {
                    Thread.sleep(2500);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
                handler1.sendMessage(msg);
            }
        }).start();





    }

    @Nullable
    @Override
    public View onCreateView(@NonNull String name, @NonNull Context context, @NonNull AttributeSet attrs) {
        return super.onCreateView(name, context, attrs);
    }
}

File: customviewsample/src/main/java/com/one/customviewsample/MainActivity.kt
package com.one.customviewsample

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.drawable.ColorDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.graphics.BitmapCompat
import androidx.core.graphics.drawable.toBitmap
import androidx.core.graphics.drawable.toDrawable

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_draw)

        val bitmap = Bitmap.createBitmap(25, 25, Bitmap.Config.ARGB_8888)
        val toDrawable = bitmap.toDrawable(resources)
        val drawable = ColorDrawable()
        val toBitmap = drawable.toBitmap(25, 25, Bitmap.Config.ARGB_8888)

    }
}
